proposed/container-configuration-meta.md

```diff
@@ -29,10 +29,70 @@ The way a framework can automatically **discover the container definitions** of
+
+In this section, we will describe the list of features that we can consider to fulfill the goal of this PSR: packages registering container entries. Let's list all possible features and later, we can vote on the features we want to keep or not.
+
+1. Ability to create a container entry using the `new` keyword.
+
+1. Ability to call methods (setters or otherwise) on a container entry.
+
+1. Ability to set public properties of a container entry.
+
+1. Ability to set private and protected properties of a container entry.
+
+1. Ability to create a container entry using a static factory method.
+
+1. Ability to create a container entry using a factory method from a container service.
+
+1. Ability to create a container entry using a closure.
+
+1. Ability to compile all container entries into a single container for maximum performance.
+
+1. Ability to alias a container entry to another.
+
+1. Ability to modify an entry defined outside of the "module" before it is returned by the container. For instance: ability for a package providing a `Twig_Extension` to modify the `Twig_Environment` by calling `$environment->addExtension($extension)`, even if the `Twig_Environment` entry was not declared by the package itself.
+
+1. Ability to create container entries for scalar values.
+
+1. Ability to create container entries for constants (from the `define` keyword or the `const` keyword).
+
+1. Ability to create container entries that are numerically indexed arrays. Values of the array can be any valid container entry (i.e. objects, scalars, another array...)
 
-### 4.2. Features
+1. Ability to create container entries that are associative arrays (maps). Values of the array can be any valid container entry (i.e. objects, scalars, another array...)
+
+1. Ability for a package to extend those arrays (add elements to the arrays). Think about a package adding a PSR-7 middleware to the list of available middlewares.
+
+1. Ability for a package to manage priority in those arrays (add a service at the beginning, at the end, in the middle, or give a priority...).
+
+1. Ability to provide a default service that should be used when binding an interface.
+
+1. Ability to declare "lazy" services (services that are wrapped into proxy objects and instanciated only when needed).
+
+1. Ability to have several services for the same class or interface (for instance, several services representing a database connections)
+
+1. Ability to declare objects that are not services (for instance, ability to manage the menu of an application through `MenuItem` container entries that would be added to a `Menu` entry).
```

Comment:
Nope... that's not what I meant... BUT, the anonymous service thing is definitely a feature we need to talk about. I will add it.

I originally opened this feature because everybody is referring to container entries as "services". But a container could very well contain objects that are not "services" (for instance a menu). I simply wanted to outline that. I realize it has more to do with "semantics" than with "features", so I'll simply remove it.